### PR TITLE
feat: add /users command

### PR DESCRIPTION
**Pull request by dibashthapa**
# Description

I added the `/users` command to list the number of connected clients

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update.

**Test Configuration**:
* OS: Manjaro
* Architecture:x86_64
* Go Version:1.17.5

# Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

**Comment by regmicmahesh** [comment]
Thanks for the pull request but unfortunately we were also working on same feature. I'd love to merge your PR if you did manage to add fetching list of users when /users in triggered isntead of just users.

P.S. Do above by merging the current dev.

File: internal/server/server.go [hunk 1]
```diff
@@ -166,6 +166,10 @@ func (s *Server) handleCommand(client *Client, cmd string, arg string) {
 		client = NewClient(client.conn, client.ipAddr)
 		s.Clients = append(s.Clients, client)
 		s.BroadcastServerMessage(fmt.Sprintf("%s joined the chat", client.username))
+	case "/users":
+		users := len(s.Clients)
+		s.SendServerPrivateMessage(fmt.Sprintf("connected users :%d", users) , client)
+
```
Comment (regmicmahesh) [inline internal/server/server.go:172]: these are outdated but sorry for late push.

**Review by regmicmahesh — CHANGES_REQUESTED** [review]

File: internal/server/server.go (hunk 1 above)
Comment (dibashthapa) [reply internal/server/server.go:172]: To my surprise, you had already refactored the whole source code just after my PR, haha.

**Comment by dibashthapa** [comment]
>  I'd love to merge your PR if you did manage to add fetching list of users when /users in triggered instead of just users.

Do you mean showing the list of users instead of no of users?
I would love to, will send PR after understanding the current source code

**Comment by regmicmahesh** [comment]
> > I'd love to merge your PR if you did manage to add fetching list of users when /users in triggered instead of just users.
> 
> Do you mean showing the list of users instead of no of users? I would love to, will send PR after understanding the current source code

yes.
something like `Users: john, jen`

**Comment by dibashthapa** [comment]
@regmicmahesh  added the changes in the new pr #15 
Sorry, I was ready to push the changes on sunday. But due to personal problems, I couldn't.

Suggest some changes in the new pr. I will update

**Comment by washbin** [comment]
Will follow up on #15

**Pull request closed without merging**